frame/treasury/src/lib.rs

```diff
@@ -255,6 +353,23 @@ decl_storage! {
 		/// Simple preimage lookup from the reason's hash to the original data. Again, has an
 		/// insecure enumerable hash since the key is guaranteed to be the result of a secure hash.
 		pub Reasons get(fn reasons): map hasher(identity) T::Hash => Option<Vec<u8>>;
+
+		/// Number of bounty proposals that have been made.
+		pub BountyCount get(fn bounty_count): BountyIndex;
+
+		/// Bounties that have been made.
+		pub Bounties get(fn bounties):
+			map hasher(twox_64_concat) BountyIndex
+			=> Option<Bounty<T::AccountId, BalanceOf<T>, T::BlockNumber>>;
+
+		/// The description of each bounty.
+		pub BountyDescriptions get(fn bounty_descriptions): map hasher(twox_64_concat) BountyIndex => Option<Vec<u8>>;
+
+		/// Bounty indices that have been approved but not yet funded.
+		pub BountyApprovals get(fn bounty_approvals): Vec<BountyIndex>;
+
+		/// Minimum value for a bounty or sub-bounty.
+		pub BountyValueMinimum get(fn bounty_value_minimum): BalanceOf<T>;
```

Comment:
This will default to zero. Not sure if that's fine or should I add a runtime migration to put some default value?

One way is to put a impossible high default, and have the council motion to place a motion to reduce it to the appropriate value

Reply:
Why would we ever want to have (allow) zero value bounties. I think there should be a lower bound. If the council is not willing to at least spend the lower bound it should not use this instrument.

That said, since the use of Bounties is governed by the Council itself there will be some conventional use in practice. Maybe there will emerge a use-case for zero value bounties. I still don’t see it however.

This decision should be driven by technical considerations (risk, state bloat, etc. ).

Reply:
5 or 10 KSM sounds like a reasonable lower bound to justify the overhead of a Bounty Instrument (management). But I guess I need to see this in practice first to really make up my mind about this.

Reply:
My current plan is add a new runtime upgrade method to initialize this to some reasonable value like 20 dollars. The exact value can be specified on the polkadot companion PR once I have done the change.

Reply:
Added https://github.com/paritytech/polkadot/pull/1336/commits/feec5a75db517e58dbdfb5b4d9a0922af6c62671

Reply:
I just came to write this issue myself. I dont see why this is a storage item vs a config value.

IMO you move it to config, and you wont have this issue. In any-case, a runtime migration is not good enough here, you must also set some default value for new chains running treasury for the first time.

Reply:
Calling `bounty_value_minimum()` should return `bounty_value_minimum.max(existential_deposit)`

Reply:
@rrtti can you find out the spec / discussion of why we want it to be a storage value?